### PR TITLE
2163 upgrade to camel K 2.1 with windows and mac tests

### PR DESCRIPTION
**Pull request by apupier**
Mixing both upgrade directly to Camel K 2.1 and providning new tests with Windows and Mac on CI because some parts depends on the others.

Add Windows and Mac platform on CI matrix for a subset of tests, basically all tests that do not require a Camel K instance.

it is fixing Language Support for standalone Java on Windows #1591

### CODE REVIEW

File: .github/workflows/OtherOSes.yml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+      - name: npm-ci
+        run: npm ci
+
+      - name: npm-vscode:prepublish
+        run: npm run vscode:prepublish
+
+      - name: test
+        id: test
+        run: npm test
+
```
Comment (djelinek) [inline .github/workflows/OtherOSes.yml:89]: we are not running UI tests for other OSes?

**Comment by djelinek** [comment]
@dependabot rebase

File: .github/workflows/OtherOSes.yml (hunk 1 above)
Comment (apupier) [reply .github/workflows/OtherOSes.yml:89]: oops

Comment (djelinek) [reply .github/workflows/OtherOSes.yml:89]: oh no, lot of failures 🙁

Comment (apupier) [reply .github/workflows/OtherOSes.yml:89]: I looked to the tests, in fact most of them are doing a deployment. On Windows and mac, i have not setup a Kubernetes instance. This is planned for another iteration. I updated teh branch again to no tplay the Ui tests on Mac and Windows for now.

**Review by djelinek — APPROVED** [review]

**Pull request merged**